File: ejercicios/c++/ejercicio059.cpp
#include <iostream>
using namespace std;

bool validador_iterativo(int numero, int base){
	int digito;
	while(numero > 0){
		digito = numero % 10;
		if(digito >= base)
			return false;
		numero /= 10;
	}
	return true;
}

int main() {
	int numero, base;
	cout << "ingrese el numero : ";
	cin >> numero;
	cout << "ingrese la base : ";
	cin >> base;
	
	if(validador_iterativo(numero, base))
		cout << "el " << numero << " es valido en base : " << base << endl;
	else
		cout << "el " << numero << " no es valido en base : " << base << endl;
	
	return 0;
}

File: ejercicios/python/ejercicio059.py
def validador_iterativo(numero, base):
   while numero > 0:
      digito = numero % 10
      numero = int(numero / 10)
      if digito >= base:
         return False
   return True

if __name__ == "__main__":
   numero = int(input("ingrese el numero : "))
   base = int(input("ingrese la base : "))
   if validador_iterativo(numero, base):
      print (str(numero) + " es valido en base " + str(base))
   else:
      print (str(numero) + " no es valido en base " + str(base))
